pkg/release/create.go

```diff
@@ -0,0 +1,99 @@
+package release
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/yaml"
+)
+
+func CreateRelease(name, base, releases, provider string, components []string, overwrite bool) error {
+	// Paths
+	baseVersion := *semver.MustParse(base) // already validated to be a valid semver string
+	providerDirectory := filepath.Join(releases, provider)
+	baseRelease, baseReleasePath, err := findRelease(providerDirectory, baseVersion)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Define release CR
+	var updatesRelease v1alpha1.Release
+	newVersion := *semver.MustParse(name) // already validated to be a valid semver string
+	updatesRelease.Name = "v" + newVersion.String()
+	for _, componentVersion := range components {
+		split := strings.Split(componentVersion, "@")
+		updatesRelease.Spec.Components = append(updatesRelease.Spec.Components, v1alpha1.ReleaseSpecComponent{
+			Name:    split[0],
+			Version: split[1],
```

Comment:
I didn't see any validation that the components have the `component@version` format so this will  panic with `slice bounds out of range` if someone forgets to provide the `@version`.

Reply:
I added some validation that apps have 2-3 parts separated by "@" symbols and components have 2 or an error is returned. Thanks for the extra push to make things nicer for users.